From e3d7208f456a67367ff6c675e28fd826284353cf Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sun, 23 Aug 2026 16:51:42 +0200
Subject: [PATCH] Remove extra Segoe UI fonts from system-ui

The Symbol font interferes with emoji presentation in some situations,
such as GitHub in Firefox. I also don't see a good reason to consider
the Historic font a "System UI" font.

Consider Segoe UI Historic as part of the sans-serif genericfamily
instead. Symbol doesn't have enough coverage to be an emoji font.

Changelog: fixed
---
 conf.d/45-latin.conf            | 8 --------
 conf.d/60-latin.conf            | 2 --
 fc-genericfamily/sans-serif.txt | 1 +
 fc-genericfamily/system-ui.txt  | 2 --
 4 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/conf.d/45-latin.conf b/conf.d/45-latin.conf
index 53158c76..b0be7855 100644
--- a/conf.d/45-latin.conf
+++ b/conf.d/45-latin.conf
@@ -297,13 +297,5 @@
 		<family>Segoe UI</family>
 		<default><family>system-ui</family></default>
 	</alias>
-	<alias>
-		<family>Segoe UI Historic</family>
-		<default><family>system-ui</family></default>
-	</alias>
-	<alias>
-		<family>Segoe UI Symbol</family>
-		<default><family>system-ui</family></default>
-	</alias>
 
 </fontconfig>
diff --git a/conf.d/60-latin.conf b/conf.d/60-latin.conf
index 6233ea28..03837901 100644
--- a/conf.d/60-latin.conf
+++ b/conf.d/60-latin.conf
@@ -81,8 +81,6 @@
 			<family>Cantarell</family>
 			<family>Noto Sans UI</family>
 			<family>Segoe UI</family>
-			<family>Segoe UI Historic</family>
-			<family>Segoe UI Symbol</family>
 			<family>Noto Sans</family><!-- fallback -->
 		</prefer>
 	</alias>
diff --git a/fc-genericfamily/sans-serif.txt b/fc-genericfamily/sans-serif.txt
index 15b196e8..4a62c08e 100644
--- a/fc-genericfamily/sans-serif.txt
+++ b/fc-genericfamily/sans-serif.txt
@@ -369,6 +369,7 @@ Raleway
 Rubik
 San Francisco
 Segoe UI
+Segoe UI Historic
 SF Pro
 SF Pro Display
 SF Pro Rounded
diff --git a/fc-genericfamily/system-ui.txt b/fc-genericfamily/system-ui.txt
index 9b369fd0..0928f078 100644
--- a/fc-genericfamily/system-ui.txt
+++ b/fc-genericfamily/system-ui.txt
@@ -32,8 +32,6 @@ Noto Sans Telugu UI
 Noto Sans Thai UI
 Noto Sans UI
 Segoe UI
-Segoe UI Historic
-Segoe UI Symbol
 Yu Gothic UI
 Khmer OS System
 Ume UI Gothic
-- 
GitLab

